src/charm.py

```diff
@@ -332,12 +333,18 @@ def _on_event(self, event, force_conflicts: bool = False) -> None:
         try:
             self._check_leader()
             self._apply_k8s_resources(force_conflicts=force_conflicts)
-            update_layer(
-                self._container_name,
-                self._container,
-                self._admission_webhook_layer,
-                self.logger,
-            )
+            self._upload_certs_to_container(event)
+            if self._certificate_files_exist():
```

Comment:
I think a better approach to this would be to call `self._upload_certs_to_container`, which could have a check inside it that verifies if the files are there, if not, try uploading them. If for any reason the push fails, we should raise an exception that can be handled in this `_on_event` method. I guess we should raise `ErrorWithStatus` or add another `except` here.

Reply:
If we were worried about refreshing certs as they expire, I'd have a problem with the updated `if certs_missing: upload_certs` since there's no way the charm ever overwrites certs.  But, we make really long lived certs atm and I don't think we're handling cert rotation atm anyway so lgtm.  

There was a nice readable quality to the previous `if certs_exist: update_layer()` that made it clear we only update layer if certs are available, but this is cool too